backend/chunker/chunker.go

```diff
@@ -0,0 +1,1412 @@
+// Package chunker provides wrappers for Fs and Object which split large files in chunks
+package chunker
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/accounting"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/fspath"
+	"github.com/rclone/rclone/fs/hash"
+)
+
+const (
+	// optimizeFirstChunk enables the following Put optimization:
+	// if a single chunk is expected, put the first chunk using base target
+	// name instead of temporary name, thus avoiding extra rename operation.
+	// WARNING: this optimization is not transaction safe!
```

Comment:
Does this mean that chunker only works on backends which support rename?

This should be in the docs...

And we should check for it in NewFs with `operations.CanServerSideMove` and return an error as soon as possible

Reply:
Will add the check. What error code is best?

Reply:
I think an informative error for the user as in "can't use chunker on a backend which doesn't support move or copy"

Reply:
Ok, errors.New("can't use chunker on a backend which doesn't support move or **rename**")

Reply:
That should say "move or copy" not "move or rename"!  Remember if the remote can't move then it is implemented as copy then delete...